### PR TITLE
chore(flow): Started adding Redux types, enabled typing on provider

### PR DESCRIPTION
**Pull request by peggyrayzis**
- Enabled Flow on the cell creator provider
- Added types lib for global types (@rgbkrk we can possibly put our generated model files here if we end up going that route)

We should consider dropping the eslint-plugin-react no-unused-prop-types rule. [It throws a false positive](https://github.com/yannickcr/eslint-plugin-react/issues/816) if you destructure your props in your component.

### CODE REVIEW

File: src/notebook/views/cell-creator.js [hunk 1]
```diff
@@ -1,19 +1,21 @@
 // @flow
+/* eslint-disable react/no-unused-prop-types */
+
 import React, { PureComponent } from 'react';
 import { throttle } from 'lodash';
 
-declare type CellCreatorProps = {|
+type Props = {
```
Comment (jdetle) [inline src/notebook/views/cell-creator.js:7]: Why the change from `declare type CellCreatorProps` to `type Props`?

**Review by jdetle — COMMENTED** [review]
Thanks for the PR, ready for merge?

File: src/notebook/views/cell-creator.js (hunk 1 above)
Comment (peggyrayzis) [reply src/notebook/views/cell-creator.js:7]: I think it was overkill for me to specify CellCreatorProps. We're checking the prop structure on the CellCreator view when the class is created so there's no reason for me to check that the {...props} being passed down in CellCreatorProvider's render method are also the same prop structure.

**Comment by peggyrayzis** [comment]
Ready for merge 😀

**Comment by rgbkrk** [comment]
> We should consider dropping the eslint-plugin-react no-unused-prop-types rule. It throws a false positive if you destructure your props in your component.

Sounds good to me!
